tests/ui/const-generics/issues/issue-113462.rs

```diff
@@ -0,0 +1,8 @@
+#[derive(Default)]
+struct S<'a, const N: S> {}
```

Comment:
```
struct S2<'b> {}

#[derive(Default)]
struct S<'a, const N: S2> {}
```

Here is a better test that has fewer unrelated cycle errors, but still reproduces the ICE (before your PR, have not tested it afterwards).

Reply:
(Updated the test to be even shorter)

Reply:
Did you see the updated comment? You should try to suppress the unused lifetime message on both structs.

Reply:
Apologies for my oversight, it should be fine now.🤦